### PR TITLE
"Don't directly throw Exception" System.Runtime.Serialization.Json

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
Contributes to #23748.
I found one place where generic Exception could reach user. I split tests because netfx throw System.Runtime.CallbackException but it doesn't exist on netcoreapp. Maybe we should add System.Runtime.CallbackException to netcore. 
Found also one helper method [here](https://github.com/MarcoRossignoli/corefx/blob/362ebe743a3d7cedb14a29a7a9f60d16545d5488/src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DiagnosticUtility.cs#L78) throws generic Exception, but i didn't found any useful test.
cc: @shmao, @zhenlan

### CODE REVIEW

File: src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs [hunk 1]
```diff
@@ -2931,7 +2931,48 @@ public static void DCJS_ValidateExceptionOnUnspecifiedRootSerializationType()
...
+                });
+
+            InvalidOperationException exception = Assert.Throws<InvalidOperationException>(() => jsonReader.Dispose());
+            Assert.IsType(typeof(DivideByZeroException), exception.InnerException);
+        }
+    }
+
+    [Fact]
+    [SkipOnTargetFramework(TargetFrameworkMonikers.Netcoreapp, "On netcoreapp throws InvalidOperationException")]
+    public static void DSJS_ThrowExceptionOnDispose_Netfx()
```
Comment (shmao) [inline src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs]: This test failed when the test was targeting UWP CoreCLR as the `SkipOnTargetFramework` above prevent the test from running on `netcoreapp` only.

We can unify the tests by checking if `innerException is InvalidOperationException || innerException.GetType().FullName == "System.Runtime.CallbackException"`

File: src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs [hunk 2]
```diff
@@ -2931,7 +2931,48 @@ public static void DCJS_ValidateExceptionOnUnspecifiedRootSerializationType()
...
+        {
+            XmlDictionaryReader jsonReader = JsonReaderWriterFactory.CreateJsonReader(ms, System.Text.Encoding.Unicode, XmlDictionaryReaderQuotas.Max,
+                reader =>
+                {
+                    //sample exception on reader close
+                    throw new DivideByZeroException();
+                });
+            try
+            {
+                jsonReader.Dispose();
```
Comment (shmao) [inline src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs]: We may want to add an `Assert.False` after the line above to make sure the above operation would throw.

File: src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs (hunk 1 above)
Comment (MarcoRossignoli) [reply src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs]: done

File: src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs (hunk 2 above)
Comment (MarcoRossignoli) [reply src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs]: done

**Comment by MarcoRossignoli** [comment]
It was a pleasure to help

**Pull request merged**